File: Day29_网络_网络入门/04_归档和解档/04_归档和解档/Person.m
//
//  Person.m
//  04_归档和解档
//
//  Created by Chiang on 16/2/25.
//  Copyright © 2016年 Apple.Inc. All rights reserved.
//

#import "Person.h"

@implementation Person

// 归档:
-(void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeObject:_name forKey:@"name"];
    [aCoder encodeInt:_age forKey:@"age"];
}

// 解档:
-(instancetype)initWithCoder:(NSCoder *)aDecoder {
    if (self = [super init]) {
        _name = [aDecoder decodeObjectForKey:@"name"];
        _age = [aDecoder decodeIntForKey:@"age"];
    }
    return self;
}

@end

File: Day34_Swift基础/09_Swift面向对象相关/01_Swift面向对象相关/Person.swift
//
//  Person.swift
//  01_Swift面向对象相关
//
//  Created by Chiang on 16/4/9.
//  Copyright © 2016年 Google. All rights reserved.
//

import UIKit

class Person: NSObject {
    var name: String?
    var age: Int?
    
    // 重写构造方法：
    override init() {
        name = "YaoMing"
        age = 35
        super.init()
    }
    
    init(dict: [String : AnyObject]) {
        setValuesForKeysWithDictionary(dict)
    }
}
